### DISCUSSION TITLE
Download photos to phone

### DISCUSSION THREAD

**Discussion by AryehGielchinsky** [immich-app/immich#6039] [Q&A] [answered]
Hello,

I'm in the process of migrating my photos from Nextcloud to Immich. I've uploaded my photos to my server from my desktop using the upload tool on the Immich web GUI. Now my photos are on my server, but not on my phone. I'd like to download the photos to my phone, which would allow me to use Immich to view them while offline.

Is there a way to use Immich to download the photos from my Immich server to my phone? This would be a one time operation, as going forward the images would be created on my phone, and then synced to my server.

Alternatively, is there a place on my phone where I can manually download the photos, and then browse them in Immich without having them get uploaded again?

Thanks

**Comment by tbleiker** [comment]
I would also like to download/add photos from shared albums or from my partner's timeline to my timeline (to have them available while offline).

**Comment by lukehmcc** [comment, marked as answer]
Afaik, if you just manually downloaded all of the photos to your phone from the server and put them in some directory that immich scans (doesn't really matter where as long as immich can see it) it should be fine. Immich checks the checksums of the image before it uploads, so if all of the check sums match the ones on your server then no duplicates should be uploaded.

It could be useful to have this as a feature in the app, but that's an easy thing to work around right now.

**Reply by aisbergde** [reply]
good to know about this useful behavior. Anyway the download could be interesting for shared albums when other users add assets.

BTW, is there some functionality, to copy assets owned by other users, contained in shared albums, into the own library?

**Comment by AryehGielchinsky** [comment]
So I ended up copying my photos to my phone from a computer, and immich did not re upload them (which is good). For any new users I guess I would copy the photos to their phone instead of using the upload GUI on the web page.

**Reply by lukehmcc** [reply]
Yeah the GUI upload is not really designed for big uploads. I have seen some successes uploading a bunch of photos that way, but the recommended way is either through the phone or the CLI.
